victim-loss-identifier/src/fetcher.ts

```diff
@@ -0,0 +1,263 @@
+        const data = (await response.json()) as coinData;
+        const price = data.coins["coingecko:ethereum"]?.price;
+
+        if (price == null) {
+          tries++; // Increment tries counter for null price
+          console.log(`Fetched price is null (attempt ${tries})`);
+        } else {
+          this.ethPriceCache.set(timestamp, price);
+          return price;
+        }
+      } catch (e) {
+        tries++;
+        console.log(`Error in fetching ETH price (attempt ${tries}):`, e);
+        if (tries === maxTries) {
+          throw e;
+        }
+      }
+
+      // Wait for 1 second before retrying
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+    }
+
+    return undefined;
+  };
+
+  private getFloorPriceInEth = async (nftAddress: string): Promise<number> => {
+    const maxTries = 3;
+    let tries = 0;
+
+    while (tries < maxTries) {
+      try {
+        const floorPrice: GetFloorPriceResponse =
+          await this.alchemy.nft.getFloorPrice(nftAddress);
+
+        const latestEntry = Object.values(floorPrice).reduce(
+          (latest: FloorPriceMarketplace | null, current) => {
+            if ("error" in current) {
+              return latest;
+            }
+
+            if (
+              !latest ||
+              new Date(current.retrievedAt) > new Date(latest.retrievedAt)
+            ) {
+              return current;
+            }
+            return latest;
+          },
+          {
+            floorPrice: 0,
+            priceCurrency: "",
+            collectionUrl: "",
+            retrievedAt: "2000-01-01T00:00:00.000Z",
+          }
+        );
+
+        if (latestEntry.priceCurrency !== "ETH") {
+          return 0;
+        }
+
+        return latestEntry.floorPrice;
+      } catch (e) {
+        tries++;
+        console.error(`Error fetching floor price (attempt ${tries}):`, e);
+        if (tries === maxTries) {
+          throw e;
+        }
+
+        // Wait for 1 second before retrying
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      }
+    }
+
+    return 0;
+  };
+
+  getNftCollectionFloorPrice = async (
+    nftAddress: string,
+    timestamp: number
+  ): Promise<number> => {
+    if (this.floorPriceCache.has(timestamp)) {
+      return this.floorPriceCache.get(timestamp)!;
+    }
+
+    const ethPrice = await this.getNativeTokenPrice(timestamp);
+    const floorPriceInEth = await this.getFloorPriceInEth(nftAddress);
+
+    if (!ethPrice || !floorPriceInEth) {
+      return 0;
+    }
+
+    const floorPrice = floorPriceInEth * ethPrice;
+    this.floorPriceCache.set(timestamp, floorPrice);
+    return floorPrice;
+  };
+
+  getErc721TransfersInvolvingScammer = async (
+    scammerAddress: string,
+    transferOccuranceTimeWindow: number
+  ): Promise<Erc721Transfer[]> => {
+    let erc721Transfers: Erc721Transfer[] = [];
+
+    const currentTime = new Date();
+    const timeWindowInMs = transferOccuranceTimeWindow * 24 * 60 * 60 * 1000;
```

Comment:
```suggestion
    const currentTime = new Date();
    const oneDayInMs = 24 * 60 * 60 * 1000;
    const timeWindowInMs = transferOccuranceTimeWindow * oneDayInMs;
```
Small suggestions to make it a bit easier to get.